src/relayserver/TransactionManager.ts

```diff
@@ -0,0 +1,208 @@
+    log.debug('gasPrice', _gasPrice)
+    log.debug('encodedCall', encodedCall)
+    const gas = parseInt(gasLimit ?? await method?.estimateGas({ from: signer }))
+    log.debug('gasLimit', gas)
+    log.debug('nonceMutex locked?', this.nonceMutex.isLocked())
+    const releaseMutex = await this.nonceMutex.acquire()
+    let signedTx
+    let storedTx: StoredTx
+    try {
+      const nonce = await this.pollNonce(signer)
+      log.debug('nonce', nonce)
+      const txToSign = new Transaction({
+        to: destination,
+        value: value,
+        gasLimit: gas,
+        gasPrice: _gasPrice,
+        data: Buffer.from(encodedCall.slice(2), 'hex'),
+        nonce
+      }, this.rawTxOptions)
+      log.trace('txToSign', txToSign)
+      // TODO omg! do not do this!
+      const keyManager = this.managerKeyManager.isSigner(signer) ? this.managerKeyManager : this.workersKeyManager
+      signedTx = keyManager.signTransaction(signer, txToSign)
+      storedTx = transactionToStoredTx(txToSign, signer, 1)
+      this.nonces[signer]++
+      await this.txStoreManager.putTx(storedTx, false)
+    } finally {
+      releaseMutex()
+    }
+    const receipt = await this.contractInteractor.sendSignedTransaction(signedTx)
+    log.debug('\ntxhash is', receipt.transactionHash)
+    if (receipt.transactionHash.toLowerCase() !== storedTx.txId.toLowerCase()) {
+      throw new Error(`txhash mismatch: from receipt: ${receipt.transactionHash} from txstore:${storedTx.txId}`)
+    }
+    return {
+      receipt,
+      signedTx
+    }
+  }
+
+  async resendTransaction (tx: StoredTx): Promise<SignedTransactionDetails> {
+    // Calculate new gas price as a % increase over the previous one
+    let newGasPrice = tx.gasPrice * retryGasPriceFactor
+    // Sanity check to ensure we are not burning all our balance in gas fees
+    if (newGasPrice > maxGasPrice) {
+      log.debug('Capping gas price to max value of', maxGasPrice)
+      newGasPrice = maxGasPrice
+    }
+    // Resend transaction with exactly the same values except for gas price
+    const txToSign = new Transaction(
+      {
+        to: tx.to,
+        gasLimit: tx.gas,
+        gasPrice: newGasPrice,
+        data: tx.data,
+        nonce: tx.nonce
+      },
+      this.rawTxOptions)
+
+    log.debug('txToSign', txToSign)
+    const keyManager = this.managerKeyManager.isSigner(tx.from) ? this.managerKeyManager : this.workersKeyManager
+    const signedTx = keyManager.signTransaction(tx.from, txToSign)
+    const storedTx = transactionToStoredTx(txToSign, tx.from, tx.attempts + 1)
+    await this.txStoreManager.putTx(storedTx, true)
+
+    log.debug('resending tx with nonce', txToSign.nonce, 'from', tx.from)
+    log.debug('account nonce', await this.contractInteractor.getTransactionCount(tx.from))
+    const receipt = await this.contractInteractor.sendSignedTransaction(signedTx)
+    log.debug('\ntxhash is', receipt.transactionHash)
```

Comment:
Maybe that should also be info when resending tx with new txhash?

Reply:
fixed